### DISCUSSION TITLE
Operating Theatre Room - Change when doctors wash their hands idea

### DISCUSSION THREAD

**Discussion by ageekhere** [CorsixTH/CorsixTH#2444] [Research Room]
Here is an idea for the Operating Theatre Room, the doctors will wash their hands when a new patient enters the room. The problem is that no matter how optimized the room is the patient will always be waiting on the doctors to be ready (when optimized for patient walk time). So the idea is have the doctors go wash their hands when an operation has finished and there is a patient in que. That way the doctors can start washing their hands while the patient walks to/in the Operating Theatre Room. By doing this you reward the player for optimizing the room layout to have the doctors be waiting on the patient rather then the other way around.

Post your ideas and thoughts below

**Comment by Alberth289346** [comment]
Since the doctors  are the bottleneck, optimising patient path looks like the wrong solution to me.

**Reply by ageekhere** [reply]
So the question is should the patient be the bottleneck or the doctors for the Operating Theatre Room? If you make the doctors start washing their hands while a patient is heading to the room instead of waiting for them to enter the room could improve this.

**Reply by Alberth289346** [reply]
We typically follow what the original program was doing. I don't know if what we currently do follows that idea, but current behavior doesn't strike me as odd.
All doctors wander around aimlessly until the patient enters. When the patient enters, the procedure begins.
That happens in all rooms, so current behavior seems consistent to me.

Assuming that the original game does that as well, I'd think that optimizing throughput wasn't a design goal for the game. To me that fits with the general idea of a non-serious fun hospital game for the general public. If you take optimization as a goal for the player, you'd get a totally different game with all kinds of performance statistics, queue lengths over time, how staff spends their time, average patient throughput, and what not.

Of course playing the current game from a performance point of view is possible and fully acceptable. One of the questions I have in my head is then: do you want to optimize the original behavior or do you want to change the game for more knobs and dials to give more tuning options? (ie change the system you're optimizing.)

One thing to watch out for here is giving too much advantage to a player. A fun game always has some pain somewhere. You have multiple paths to get good behavior, but none of them is a true winner. As a result you keep trying to find improvements, and you keep playing with other setups. If the game would provide a clear true winning strategy for every room, everybody simply picks that option and all your discussion and thinking about alternatives with pros and cons just dies.

**Comment by ARGAMX** [comment]
> Here is an idea for the Operating Theatre Room, the doctors will wash their hands when a new patient enters the room. The problem is that no matter how optimized the room is the patient will always be waiting on the doctors to be ready (when optimized for patient walk time). So the idea is have the doctors go wash their hands when an operation has finished and there is a patient in que. That way the doctors can start washing their hands while the patient walks to/in the Operating Theatre Room. By doing this you reward the player for optimizing the room layout to have the doctors be waiting on the patient rather then the other way around.
> 
> Post your ideas and thoughts below

An interesting and clever idea, but it's a deviation from the vanilla TH, in which surgeons always start washing hands after the next patient enters.
